src/librustc_span/source_map.rs

```diff
@@ -715,7 +715,7 @@ impl SourceMap {
         }
     }
 
-    pub fn def_span(&self, sp: Span) -> Span {
+    pub fn guess_head_span(&self, sp: Span) -> Span {
         self.span_until_char(sp, '{')
```

Comment:
This could use a FIXME for improving it, or maybe for replacing it with a span computed by the parser and kept in e.g. `{ast,hir}::Item`. But maybe not in this PR, since I just want to get this out of the way.

Reply:
However, having a doc comment explaining the caveats of the "guessing" would be really useful for those of us who have no idea...

Reply:
It could say that it's a diagnostic-only hack, I suppose (e.g. to shrink the `Span` to not refer to an entire e.g. item, but just mainly the `fn foo` or `struct Bar` part).

The caveat is it can't possibly work in all cases and instead we should track a relevant `Span` throughout compilation (or maybe even reuse the one in the `Ident`, i.e. point to just the name).

Reply:
Yeah, something like that would be useful as a doc comment. :)

Reply:
@estebank would be nice if you could add that when you have to rebase anyway.